File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.py
def solution(n):
    answer = 0
    if n == 1:
        return 1
    if n == 2:
        return 2
    # 1칸이나 2칸을 뛸 수 있음
    dp = [0] * (n+1)
    dp[1] = 1
    dp[2] = 2
    for i in range(3, n+1):
        dp[i] = dp[i-1] + dp[i-2]
    ans = dp[-1] % 1234567
    return ans

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.js
function solution(n) {
    if (n === 1) {
        return 1;
    } else if (n === 2) {
        return 2;
    }
    
    var dp = [];
    dp[0] = 0;
    dp[1] = 1;
    dp[2] = 2;
    
    for (i = 3; i <= n; i++) {
        dp[i] = (dp[i - 1] + dp[i - 2]) % 1234567; // DP를 사용하여 경우의 수 계산
    }
    console.log(dp)
    return dp[n];
}